lib/ansible/module_utils/facts/system/systemd.py

```diff
@@ -0,0 +1,42 @@
+# Get systemd version and features
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import ansible.module_utils.compat.typing as t
+
+from ansible.module_utils.facts.collector import BaseFactCollector
+from ansible.module_utils.facts.system.service_mgr import ServiceMgrFactCollector
+
+
+class SystemdFactCollector(BaseFactCollector):
+    name = "systemd"
+    _fact_ids = set()  # type: t.Set[str]
+
+    def collect(self, module=None, collected_facts=None):
+        systemctl_bin = module.get_bin_path("systemctl")
+        if systemctl_bin and ServiceMgrFactCollector.is_systemd_managed(module):
+            rc, stdout, stderr = module.run_command(
+                [systemctl_bin, "--version"],
+                check_rc=True,
```

Comment:
fact gathering are 'best effort', you should not fail the whole collection here. You can alternatively issue a warning.

Reply:
would the following be acceptable?
`ServiceMgrFactCollector.is_systemd_managed(module)` is used so we don't fail on systems that doesn't use `systemd`, `if systemctl_bin` is used since I'm already fetching the path.

```diff
diff --git a/lib/ansible/module_utils/facts/system/systemd.py b/lib/ansible/module_utils/facts/system/systemd.py
index 6b3fcee971..bbacbbe6de 100644
--- a/lib/ansible/module_utils/facts/system/systemd.py
+++ b/lib/ansible/module_utils/facts/system/systemd.py
@@ -28,6 +28,9 @@ class SystemdFactCollector(BaseFactCollector):
     _fact_ids = set()  # type: t.Set[str]
 
     def collect(self, module=None, collected_facts=None):
+        systemd_facts = {}
+        systemd_facts["systemd"] = {}
+
         systemctl_bin = module.get_bin_path("systemctl")
         if systemctl_bin and ServiceMgrFactCollector.is_systemd_managed(module):
             rc, stdout, stderr = module.run_command(
@@ -35,8 +38,7 @@ class SystemdFactCollector(BaseFactCollector):
                 check_rc=True,
             )
 
-            systemd_facts = {}
-            systemd_facts["systemd"] = {}
             systemd_facts["systemd"]["features"] = str(stdout.split("\n")[1])
             systemd_facts["systemd"]["version"] = int(stdout.split(" ")[1])
-            return systemd_facts
+
+        return systemd_facts
```

Reply:
the issue is `check_rc=True` as if the command fails for any reason (permissions/resources/etc) it will kill the rest of fact gathering

Reply:
ah yes, the things i didn't think of.

Reply:
no worries, that is why we have reviews ... and we still miss stuff